### PR TITLE
config: extending usage of EnvoyMobileHttpConnectionManager

### PR DESCRIPTION
**Pull request by alyssawilk**
https://github.com/envoyproxy/envoy/pull/17110 the EnvoyMobileHttpConnectionManager but it turns out it's instantiated either as a standard network filter (needs a name) or as an API listener (which only accepted HttpConnectionManager)

Fixing both issues.

Risk Level: low
Testing: unit tests
Docs Changes: n/a
Release Notes: n/a
Part of https://github.com/envoyproxy/envoy-mobile/issues/1540

### CODE REVIEW

**Comment by alyssawilk** [comment]
:-(

/wait

File: source/server/api_listener_impl.cc [hunk 1]
```diff
@@ -32,13 +32,25 @@ void ApiListenerImplBase::SyntheticReadCallbacks::SyntheticConnection::raiseConn
...
+          EnvoyMobileHttpConnectionManager>(config.api_listener().api_listener(),
+                                            factory_context_.messageValidationVisitor());
 
   http_connection_manager_factory_ = Envoy::Extensions::NetworkFilters::HttpConnectionManager::
       HttpConnectionManagerFactory::createHttpConnectionManagerFactoryFromProto(
-          typed_config, factory_context_, read_callbacks_);
+          typed_config.config(), factory_context_, read_callbacks_, false);
+  END_TRY
+  catch (const EnvoyException& e) {
+    auto typed_config = MessageUtil::anyConvertAndValidate<
```
Comment (mattklein123) [inline source/server/api_listener_impl.cc:46]: nit: this is a bit odd to retry during the exception catch. Can you potentially just match on the type_url of the config and then have a better not supported message if it's neither of the supported ones? Or we could do a specific registry for things that work with the API listener? At minimum I would do a comment/TODO here.

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Thanks LGTM with small comment.

/wait

File: source/server/api_listener_impl.cc (hunk 1 above)
Comment (alyssawilk) [reply source/server/api_listener_impl.cc:46]: would you accept a tracking bug and a follow-up PR?
I'd prefer to make sure this all hangs together on the mobile side (as the first round didn't) and adding a TODO would require another CI cycle.

Comment (alyssawilk) [reply source/server/api_listener_impl.cc:46]: https://github.com/envoyproxy/envoy/issues/17434 filed optimistically but no pressure :-)

Comment (mattklein123) [reply source/server/api_listener_impl.cc:46]: Sure a follow up is fine, thanks.

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**